src/db.rs

```diff
@@ -1544,6 +1544,76 @@ impl<T: ThreadMode, D: DBInner> DBCommon<T, D> {
         }
     }
 
+    /// Set the database entry for "key" to "value" with WriteOptions.
+    /// If "key" already exists, it will be overwritten.
+    /// Takes an additional argument `ts` as the timestamp.
+    /// Note: the DB must be opened with user defined timestamp enabled.
+    pub fn put_with_ts_opt<K, V, S>(
+        &self,
+        key: K,
+        ts: S,
+        value: V,
+        writeopts: &WriteOptions,
+    ) -> Result<(), Error>
+    where
+        K: AsRef<[u8]>,
+        V: AsRef<[u8]>,
+        S: AsRef<[u8]>,
+    {
+        let key = key.as_ref();
+        let value = value.as_ref();
+        let ts = ts.as_ref();
+        unsafe {
+            ffi_try!(ffi::rocksdb_put_with_ts(
+                self.inner.inner(),
+                writeopts.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                ts.as_ptr() as *const c_char,
+                ts.len() as size_t,
+                value.as_ptr() as *const c_char,
+                value.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Put with timestamp in a specific column family with WriteOptions.
+    /// If "key" already exists, it will be overwritten.
+    /// Takes an additional argument `ts` as the timestamp.
```

Comment:
what happens if we have key_1, ts_1 and we try to write key_1, ts_2. Does ts_2 replace ts_1 or do we have two kvs now?

Reply:
It should be coexists. Thanks, I'll add more clarification.